IDE/Renesas/e2studio/RX72N/Readme.md

```diff
@@ -0,0 +1,303 @@
+They are compiled by e2Studio and running on the target board.
+
+In this demo, you may download two versions of application binary file by Renesas Flash Programmer.
+You can download and excute wolfBoot by e2Studio debugger. Use a USB connection between PC and the
+board for the debugger and flash programmer.
+
+## 2. Components and Tools
+
+|Item|Name/Version|Note|
+|:--|:--|:--|
+|Board|RX72N/Envision Kit||
+|MCU|Renesas RX72N|R5F572NNxFB|
+|IDE|e2studio 2022-07|Download from Renesas site|
+|Compiler|CCRX v3.04.00||
+|FIT Module||Download from Renesas site|
+|Flash Writer|Renesas Flash Programmer v3|Download from Renesas site|
+|Key tools|keygen and sign|Included in wolfBoot|
+|rx-elf-objcopy|GCC for Renesas RX 8.3.0.202202-GNURX-ELF|Included in GCC for Renesas RX|
+
+
+FIT Module
+|Module|Version|Note|
+|:--|:--|:--|
+|r_bsp|v7.20|#define BSP_CFG_USTACK_BYTES            (0x2000)|
+||key size uses rsa-3072, please sets to (0x3000)|
+|r_flash_rx|v4.90||
+
+
+Flash Allocation:
+```
++---------------------------+------------------------+-----+
+| B |H|                     |H|                      |     |
+| o |e|   Primary           |e|   Update             |Swap |
+| o |a|   Partition         |a|   Partition          |Sect |
+| t |d|                     |d|                      |     |
++---------------------------+------------------------+-----+
+0xffc00000: wolfBoot
+0xffc10000: Primary partition (Header)
+0xffc10100: Primary partition (Application image) /* When it uses IMAGE_HEADER_SIZE 256, e.g. ED25519, EC256, EC384 or EC512 */
+0xffc10200: Primary partition (Application image) /* When it uses IMAGE_HEADER_SIZE 512, e.g. RSA2048, RSA3072 */
+0xffdf0000: Update  partition (Header)
+0xffdf0100: Update  partition (Application image)
+0xfffd0000: Swap sector
+
+```
+
+Note : Depending on IMAGE_HEADER_SIZE, it needs to change the address of Power Reset vector by Linker section.
+Application default is set to 0xffc10200. It means that you need to change it when you use 256 IMAGE_HEADER_SIZE.
+
+
+## 3. How to build and use
+It has key tools running under the host environment such as Linux, Windows or MacOS.
+For comiling the tools, follow the instruction described in the user manual.
+
+It demonstrates simple secure firmware update by wolfBoot. A sample application v1 is
+cerurely updated to v2. Both versions behave the same except displaying its version of v1 or v2.
+They are compiled by e2Studio and running on the target board.
+
+In this demo, you may download two versions of application binary file by Renesas Flash Programmer.
+You can download and excute wolfBoot by e2Studio debugger. Use a USB connection between PC and the
+board for the debugger and flash programmer.
+
+
+### 3-1 Key generation
+
+```
+$ cd <wolfBoot>
+$ export PATH:$PATH:<wolfBoot>/tools/keytools
+$ keygen --ecc256 -g ./pri-ecc256.der
+```
+
+It generates a pair of private and public key with -g option. The private key is stored 
```

Comment:
- suggest: "This generates a pair [...]" (or "This command generates one pair [...]")
- "of private and public keys" or "of one public and one private keys"